crates/blockchain/dev/block_producer.rs

```diff
@@ -29,9 +29,10 @@ pub async fn start_block_producer(
 
         let payload_attributes = PayloadAttributesV3 {
             timestamp: SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs(),
+            prev_randao: H256::zero(),
             suggested_fee_recipient: coinbase_address,
             parent_beacon_block_root: Some(parent_beacon_block_root),
-            ..Default::default()
```

Comment:
I think we need to change `PayloadAttributesV3 ` to not take an Option

Reply:
Yes, it could work, but we've changed it to be an `Option` in PR #1510 and we use `PayloadAttributesV3` in all the `ForkChoiceUpdatedV*` structs we have:

```rust
pub struct ForkChoiceUpdatedV1 {
    pub fork_choice_state: ForkChoiceState,
    pub payload_attributes: Option<PayloadAttributesV3>,
}
#[derive(Debug)]
pub struct ForkChoiceUpdatedV2 {
    pub fork_choice_state: ForkChoiceState,
    pub payload_attributes: Option<PayloadAttributesV3>,
}
#[derive(Debug)]
pub struct ForkChoiceUpdatedV3 {
    pub fork_choice_state: ForkChoiceState,
    pub payload_attributes: Option<PayloadAttributesV3>,
}
```

Changing it back to `pub withdrawals: Vec<Withdrawal>` will need some further changes right? I can do it if they are essential.

Reply:
Yeah, I had to change it recently to have V1 to work. We could implement `PayloadAttributesV2` and `PayloadAttributesV1` eventually, but that can create a parsing nightmare as there are tests calling V1 with V2 or V3 attrs (and the other way around) that expect specific errors... So it was easier to have a single parser for all 3 versions.
We can create a ticket to look at that later, though

Reply:
sounds good